.github/workflows/lint.yml

```diff
@@ -0,0 +1,35 @@
+name: Lint
+
+on: [pull_request]
+
+jobs:
+  misspell:
+    name: Check Spelling Misspell All Files In Commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install
+        run: wget -O - -q https://git.io/misspell | sh -s -- -b .
+      - name: Misspell
+        run: git ls-files --empty-directory | xargs ./misspell -error
+  pre-commit:
```

Comment:
pre commits are supposed to be `pre-commits` :D I don't think there should be any stage in github actions.